src/services/smart-fields-values-injector.js

```diff
@@ -68,16 +77,22 @@ function SmartFieldsValuesInjector(record, modelName, fieldsPerModel, depth = 0)
 
   this.perform = () =>
     P.each(schema.fields, (field) => {
-      if (!record[field.field]) {
+      if (record
+        && record.dataValues
+        && !Object.prototype.hasOwnProperty.call(record.dataValues, field.field)) {
         if (field.get || field.value) {
-          if (isNotRequestedField(modelName, field.field)) {
+          if (isNotRequestedField(requestedField || modelName, field.field)) {
             return null;
           }
 
           return setSmartFieldValue(record, field, modelName);
         }
         if (_.isArray(field.type)) {
-          record[field.field] = [];
+          // NOTICE: when modifying a sequelize record, make sure you pass the same reference
+          //         to record and record.dataValues
+          const emptyArray = [];
+          record[field.field] = emptyArray;
+          record.dataValues[field.field] = emptyArray;
```

Comment:
This part of code enlightens the fix

Reply:
I clearly don't think we should use `dataValues` that much. It's way too dangerous, it can breaks by just updating a minor version

Reply:
From the moment, we manipulate and modifiy directly sequelize object, we need to reflect added attributes in `dataValues` and direct attributes. Otherwise a lot of hard bugs could happen, if one is using [`getDataValue`](https://github.com/sequelize/sequelize/blob/a2d75e6edd477b6167636d2a7e446346d666b9d9/src/model.js#L3658)
https://sequelize.readthedocs.io/en/v3/api/instance/

Reply:
In that case you should use the public function `setDataValue(key, value)`

Reply:
Thanks I will update that ;)